src/modules/Popup/Popup.js

```diff
@@ -188,15 +192,25 @@ export default class Popup extends Component {
       }
     }
 
-    if (offset) {
-      if (_.isNumber(style.right)) {
-        style.right -= offset
-      } else {
-        style.left -= offset
-      }
+    return { ...style, ...this.applyOffset(style) }
+  }
+
+  applyOffset = (style) => {
+    const { horizontalOffset, verticalOffset } = this.props
+    return {
+      ...this.applyHorizontalOffset(style, horizontalOffset),
+      ...this.applyVeriticalOffset(style, verticalOffset),
     }
+  }
+
+  applyVeriticalOffset = ({ top, bottom }, offset) => {
+    if (_.isNumber(top)) return { bottom, top: top + offset }
+    return { top, bottom: bottom + offset }
```

Comment:
@bharatzvm, the`verticalOffset` calculation should be different based on if the popup is located above or below the _trigger_.  Are you sure this logic will not work as expected when the popup is below the trigger?

Reply:
I followed the notion of moving the popup outwards of the screen through the centre of the screen from the explicitly positioned direction wherever it is. It was also one of the reason why I had to add the offset.

Reply:
@adam-26 My bad again, I was wrong thanks for spotting it.

Reply:
@adam-26 I think the logic works fine irrespective of the positioning because if it can't place the popup in the view port it tries all other positions until it find ones in the viewport if it cant, it falls back to the position user specified position. I will be updating the pr to fix these reviews